### PR TITLE
Improve error reporting for missing members

### PR DESCRIPTION
**Pull request by hamzaremmal**
Fixes: #19731

### CODE REVIEW

**Comment by som-snytt** [comment]
orthogonal note to self: scala 2 has improved output that I ported at some point but apparently I did not PR it.

**Comment by hamzaremmal** [comment]
Some tests timed out, re-running the CI.

File: compiler/src/dotty/tools/dotc/typer/RefChecks.scala [hunk 1]
```diff
@@ -654,8 +654,9 @@ object RefChecks {
 
         val missingMethods = grouped.toList flatMap {
           case (name, syms) =>
-            val withoutSetters = syms filterNot (_.isSetter)
-            if (withoutSetters.nonEmpty) withoutSetters else syms
+            syms.filterConserve(! _.isSetter)
+            .groupBy(_.signature) // Avoid duplication for similar definitions (#19731)
+            .map(f => f._2.head) // _1: Closest undefined method to the scope
```
[the remark this answers is not in the document]
Comment (hamzaremmal) [reply compiler/src/dotty/tools/dotc/typer/RefChecks.scala:659]: Do I have the same guarantees as groupBy ? The order of the symbols for each key is important.

[the remark this answers is not in the document]
Comment (sjrd) [reply compiler/src/dotty/tools/dotc/typer/RefChecks.scala]: `distinct` and `distinctBy` both select the first element among elements considered the same, and preserve the relative order of the elements they do select. That's as good guarantees as one can hope for.

**Review by sjrd — APPROVED** [review]

**Pull request merged**